lsコマンドを作る2

pofkuma opened the pull request:
`-a`オプション付きのlsコマンドを作成しました。

ご確認よろしくお願いします。

pofkuma committed:
Make "-a" option available

"-a" オプションでドット始まりのファイルも表示できるようにする

pofkuma committed:
Add tests and Make method testable

"-a"オプションのテストができるように修正

pofkuma committed:
Use shoter logic for calculate line_count

表示行数の計算方法を短くする
(割り算の結果を小数として出して、小数点を切り上げ)
https://github.com/pofkuma/ruby-practices/pull/6#discussion_r755137515

summertree128 commented on 05.ls/ls.rb:
```diff
@@ -21,16 +15,30 @@ def convert_layout(file_names, line_count, max_name_length)
-  line_count = calculate_line_count(contents.size)
+  return if contents.size.zero?
+
+  line_count = (contents.size / MAX_COLUMUNS.to_f).ceil
```
👍

summertree128 commented on 05.ls/ls.rb:
```diff
@@ -21,16 +15,30 @@ def convert_layout(file_names, line_count, max_name_length)
-def list_directory_contents(contents)
-  format(contents.sort) unless contents.size.zero?
+def list_directory_contents(options)
+  entries =
+    if options[:a]
+      Dir.glob('*', File::FNM_DOTMATCH)
+    else
+      Dir.glob('*')
+    end
```
そのままでもいいですし、三項演算子を使うと一行でも書けますね。僕だったら一行で書くような気がします。
参考：https://qiita.com/lasershow/items/160c854e4256ba596ec5

pofkuma replied:
三項演算子を試してみたのですが、なんとなく` Dir.glob`が縦に揃っている方が好みだったのでそのままにします。
教えてくださりありがとうございます！

summertree128 commented on 05.ls/ls_test.rb:
```diff
@@ -6,31 +6,31 @@
   def test_ls_no_options_no_contents
     content_name_lists = []
-    assert_nil list_directory_contents(content_name_lists)
+    assert_nil format(content_name_lists)
```
`format`メソッドをテストするケースと`list_directory_contents`が混在しているので、この区別はあった方がいいかもしれませんね。例えば`format`をテストする方はテストメソッド名を`test_format_*`とすると分かりやすいかもです。

pofkuma replied:
テストメソッド名まで気が回っていませんでした…。ありがとうございます🙏
https://github.com/pofkuma/ruby-practices/pull/7/commits/3a52822649b7b2ef9d738370f27762f7dca96fa2 で対応いたしました。

pofkuma committed:
Modify test method names

テスト対象のメソッドに合わせて名前を修正
https://github.com/pofkuma/ruby-practices/pull/7#discussion_r757884989
